### PR TITLE
[project-base] improved frontend performance

### PR DESCRIPTION
**Pull request by grossmannmartin**
- js files are now rendered at the bottom, so it's not blocking page anymore
- added customized (smaller) jQueryUI library

| Q             | A
| ------------- | ---
|Description, reason for the PR| 
|New feature| No 
|BC breaks| No 
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by vitek-rostislav** [comment]
I promised to link the old pull request for JS minification, so here it is (might not be relevant here anymore :laughing:) https://git.shopsys-framework.com/shopsys/shopsys-framework/merge_requests/10

**Comment by boris-brtan** [comment]
https://developers.google.com/speed/pagespeed/insights/?url=

File: UPGRADE.md [hunk 1]
```diff
@@ -84,6 +84,12 @@ There is a list of all the repositories maintained by monorepo, changes in log b
     - modify your Redis configuration (`app/config/packages/snc_redis.yml`) by prefixing all the prefix values with the value of the env variable (`%env(REDIS_PREFIX)%`)
 - [#679 webserver container starts after php-fpm is started](https://github.com/shopsys/shopsys/pull/679)
     - add `depends_on: php-fpm` into `webserver` service of your `docker-compose.yml` file so webserver will not fail on error `host not found in upstream php-fpm:9000`
+- *(optional)* [#703 improved frontend performace](https://github.com/shopsys/shopsys/pull/703)
+    - in your `base.html.twig` template move non-essential javascript files at the bottom of a page.
+    We cannot provide exact instructions as we don't know your implementation.
+    For an example take a look at [changes](https://github.com/shopsys/shopsys/pull/703/files#diff-4c948fb55a9ceba2f3070e572ac506f3)
+    - you can use new simplified jquery-ui.min.js file
+    - please be aware, that if your implementation use more component from jQueryUI, you have to be extra careful
```
Comment (boris-brtan) [inline UPGRADE.md]:
```suggestion
    - please be aware that if your implementation uses more components from jQueryUI, you have to be extra careful
```

File: UPGRADE.md [hunk 2]
```diff
@@ -84,6 +84,12 @@ There is a list of all the repositories maintained by monorepo, changes in log b
     - modify your Redis configuration (`app/config/packages/snc_redis.yml`) by prefixing all the prefix values with the value of the env variable (`%env(REDIS_PREFIX)%`)
 - [#679 webserver container starts after php-fpm is started](https://github.com/shopsys/shopsys/pull/679)
     - add `depends_on: php-fpm` into `webserver` service of your `docker-compose.yml` file so webserver will not fail on error `host not found in upstream php-fpm:9000`
+- *(optional)* [#703 improved frontend performace](https://github.com/shopsys/shopsys/pull/703)
+    - in your `base.html.twig` template move non-essential javascript files at the bottom of a page.
```
Comment (boris-brtan) [inline UPGRADE.md]: OK, maybe if this is just optional we do not need to list essential and non-essential javascript files

File: UPGRADE.md [hunk 3]
```diff
@@ -84,6 +84,12 @@ There is a list of all the repositories maintained by monorepo, changes in log b
...
+    We cannot provide exact instructions as we don't know your implementation.
+    For an example take a look at [changes](https://github.com/shopsys/shopsys/pull/703/files#diff-4c948fb55a9ceba2f3070e572ac506f3)
```
Comment (boris-brtan) [inline UPGRADE.md]: please check whether the hash will work after fixups will be squashed

**Comment by vitek-rostislav** [comment]
Hi guys, I noticed sometimes there is a quite significant amount of time until the honeypot on contact form is hidden :confused: 

I encounter this only for the first page load.

image

**Comment by grossmannmartin** [comment]
> Hi guys, I noticed sometimes there is a quite significant amount of time until the honeypot on contact form is hidden 😕
> 
> I encounter this only for the first page load.
> 
> image

Hi @vitek-rostislav,
you're right. I try to test it and it seems like a real problem.

Video of behavior follows
https://drive.google.com/file/d/1K6SftTFMN5k28V_RBfyBdqfgvQL49wJN/view?usp=sharing

Can you, @tomshopsys, take a look at it, please?

**Comment by tomshopsys** [comment]
I have just added class display-none to hide form by default - but I am thinking about if it looses form functionality (fill form security) or not. Can you think about it too?

**Comment by vitek-rostislav** [comment]
>I have just added class display-none to hide form by default - but I am thinking about if it looses form functionality (fill form security) or not. Can you think about it too?

Hi, I think it will work, however, it is probably less secure :confused: 
https://stackoverflow.com/questions/36227376/better-honeypot-implementation-form-anti-spam

**Comment by grossmannmartin** [comment]
> > I have just added class display-none to hide form by default - but I am thinking about if it looses form functionality (fill form security) or not. Can you think about it too?
> 
> Hi, I think it will work, however, it is probably less secure 😕
> https://stackoverflow.com/questions/36227376/better-honeypot-implementation-form-anti-spam
> 
> We will discuss this...

Thank you, @vitek-rostislav 
in that case, I'm for accepting this solution and create a separate issue (UserStory) and make a deeper analysis/resolution of honeypot there.

**Comment by PatrikBajer** [comment]
Hey,

I get it for test on http://tg-js.heimdall.netdevelo:7980

1) footer newsletter -> I write "a" to input for email and try send it with that email. After first send I get notification about incorrect email. I ignore it and send it again - after that page drop to error 400 with URL - http://tg-js.heimdall.netdevelo:7980/newsletter/subscribe-email/

2) Contact us form on homepage - http://prntscr.com/m7fbo0 - I write to email input incorrect email and send it. There is not warning message.

3) Same problem like point 1) - http://tg-js.heimdall.netdevelo:7980/contactForm/ after 2nd send form drop. 

4) I don't like controling of email. (I think it is for new issues) because we are verifyin email after user sent form. In practice - Go to registration/3step order fill incorrect email and after that send form. He says you about incorrect email address but if you fill that input with right email. Its still showing you incorrect email. (but it isn't right at that point) - http://prntscr.com/m7g0i1

**Comment by PetrHeinz** [comment]
@grossmannmartin Just so you now, I have some work-in-progress PR to fix the issues regarding JS validation in #648, not sure whether it fixes this particular problem though. Might be helpful if you or Patrik decide to look into it deeper :slightly_smiling_face:

**Pull request merged**